### PR TITLE
Remove unnecessary installations

### PR DESCRIPTION
**Pull request by mttrms**
In its current state, the laptop script installs too many things that are not necessary to run our core services. It's also highly opinionated in certain configs / preferences.

This has lead to a few issues with new developers:
* **Not** running the script (and missing core services) because it does too much
* Running the script but with unwanted side effects
* Having to edit the script before running it

This is the first pass at simplifying our installs. It's not perfect, but does remove all unnecessary installations. Follow up PRs to come for some further improvements.

### CODE REVIEW

File: mac.sh [hunk 1]
```diff
@@ -242,66 +194,23 @@ install_postgresql
...
 brew_install_or_upgrade 'rbenv'
 brew_install_or_upgrade 'ruby-build'
-brew_install_or_upgrade 'wget'
-brew_install_or_upgrade 'autojump'
 brew_install_or_upgrade 'openssl'
 brew unlink openssl && brew link openssl --force
-brew_install_or_upgrade 'z'
 
 brew_tap 'homebrew/cask'
-cask_install_or_upgrade 'chromedriver'
```
Comment (jmcolella) [inline mac.sh:263]: is this an opinionated install? i thought this was required for running certain specs

File: mac.sh [hunk 2]
```diff
@@ -242,66 +194,23 @@ install_postgresql
 brew_install_or_upgrade 'redis'
 brew_launchctl_restart 'redis'
 brew_install_or_upgrade 'sops'
-brew_install_or_upgrade 'the_silver_searcher'
-brew_install_or_upgrade 'vim'
```
Comment (jmcolella) [inline mac.sh:246]: did you consider adding smaller scripts for folks to opt-in to if they want some tools installed, like `vim` or `oh_my_zsh`?

File: mac.sh (hunk 1 above)
Comment (magcurt8) [reply mac.sh:263]: yeah sadly this is required if you want to run the feature spec suite locally for the monolith :/

Comment (mttrms) [reply mac.sh:263]: ah yeah, for some reason I thought we removed most / all of that but will add back in

File: mac.sh (hunk 2 above)
Comment (mttrms) [reply mac.sh:246]: yeah I talked through this with @showell215 @gummoe @jsonunger - but we all agreed most of that would come based on what someone personally likes or sees in a team buddies setup.

I would like to make the install a little more interactive or choice-based. I think when that comes, we could add in things like node/ruby version manager choice, vim / nvim / vscode, oh my zsh, various plugins, etc.

File: mac.sh [hunk 3]
```diff
@@ -161,40 +152,6 @@ install_bundler_1_17_3() {
   bundle config --global jobs $((number_of_cores - 1))
 }
 
-install_vim_config() {
```
Comment (victorwhy) [inline mac.sh:164]: wow matt, big of you to remove `vim`

File: mac.sh (hunk 2 above)
Comment (jmcolella) [reply mac.sh:246]: Got it, yeah interaction-based would be nice. Sounds good.

I wonder, though, if something like core language version managers should be standardized so it's easier for folks to communicate when issues arise. Just my take, not massively opinionated on that.

File: mac.sh (hunk 3 above)
Comment (mttrms) [reply mac.sh:164]: I know that anyone that wants vim has been hand crafting the perfect config for a long time :)

Comment (showell215) [reply mac.sh:164]: Matt doesn't even use a config, he's pure vanilla vim

File: mac.sh (hunk 1 above)
Comment (magcurt8) [reply mac.sh:263]: someday @mttrms, someday </3

File: README.md [hunk 4]
```diff
@@ -94,93 +94,22 @@ What it sets up
 ---------------
 
 * [Bundler] for managing Ruby libraries
-* [Exuberant Ctags] for indexing files for vim tab completion
-* [Foreman] for managing web processes
```
Comment (magcurt8) [inline README.md:98]: I think people still use foreman?

Comment (mttrms) [reply README.md:98]: fun fact this wasn't actually installing foreman! just readme cleanup, but the `pg` `bin/setup` (or some setup file) does install foreman there

File: mac.sh [hunk 5]
```diff
@@ -242,66 +194,24 @@ install_postgresql
...
 brew_install_or_upgrade 'openssl'
 brew unlink openssl && brew link openssl --force
-brew_install_or_upgrade 'z'
 
 brew_tap 'homebrew/cask'
 cask_install_or_upgrade 'chromedriver'
 install_elasticsearch
 
 # Install applications
-cask_install_or_upgrade 'google-chrome'
```
Comment (magcurt8) [inline mac.sh:267]: I think actually... we might need it for the feature specs.. merp

File: mac.sh [hunk 6]
```diff
@@ -242,66 +194,24 @@ install_postgresql
...
 cask_install_or_upgrade 'google-cloud-sdk'
 gcloud components install kubectl docker-credential-gcr pubsub-emulator
 brew_install_or_upgrade 'helm'
 append_to_zshrc 'source /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/path.zsh.inc'
-append_to_zshrc 'source /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/completion.zsh.inc'
-append_to_zshrc 'eval "$(kubectl completion zsh)"'
-append_to_zshrc 'eval "$(helm completion zsh)"'
-
-# Configure git aliases
-git config --global alias.st status
```
Comment (magcurt8) [inline mac.sh:285]: awwwww yisss

**Review by magcurt8 — COMMENTED** [review]
this lgtm overall, just a few libs that need to be kept for feature specs (sadly)

File: README.md (hunk 4 above)
Comment (magcurt8) [reply README.md:98]: bahahah nice. good catch!

File: mac.sh (hunk 2 above)
Comment (jsonunger) [reply mac.sh:246]: I love the idea of an interactive installer. When it comes to things like version managers, I originally felt similarly that it should be standardized, but the reality is that so long as you use the right version in whatever repo you're in, I don't see why we need to prescribe a specific manager "globally".

**Review by maxnovak — APPROVED** [review]
MTᎮ⅃
Love seeing all that red 🔥

**Review by gummoe — APPROVED** [review]
aside from any of the 'do we need this for e2e tests' considerations, I say that this ia GREAT to go

File: mac.sh (hunk 5 above)
Comment (mttrms) [reply mac.sh:267]: chrome is installed by default on our laptops so I don't think this line is required - but I can add it back anyway

Comment (magcurt8) [reply mac.sh:267]: oh right! my b, nah leave it out, I forgot that slack + chrome and stuff are installed by IT before we get our laptops now (we used to install it ourselves)

**Pull request merged**